bike-rental-be/src/main/java/lt/psk/bikerental/controller/CheckController.java

```diff
@@ -0,0 +1,58 @@
+package lt.psk.bikerental.controller;
+
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lt.psk.bikerental.DTO.Check.CheckDTO;
+import lt.psk.bikerental.entity.Check;
+import lt.psk.bikerental.entity.User;
+import lt.psk.bikerental.repository.CheckRepository;
+import lt.psk.bikerental.repository.UserRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/check")
+@RequiredArgsConstructor
+public class CheckController {
+
+    private final CheckRepository checkRepository;
+    private final UserRepository userRepository;
+    private final ModelMapper modelMapper;
+
+    @GetMapping
+    public ResponseEntity<List<CheckDTO>> getAllChecks(@AuthenticationPrincipal Jwt jwt) {
+        UUID userId = getUserIdFromJwt(jwt);
+
+        List<CheckDTO> checks = checkRepository.findAllByUserId(userId, Sort.by(Sort.Direction.DESC, "createdAt")).stream()
+                .map(check -> modelMapper.map(check, CheckDTO.class))
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(checks);
+    }
+
+    @GetMapping("/latest")
+    public ResponseEntity<CheckDTO> getLatestCheck(@AuthenticationPrincipal Jwt jwt) {
+        UUID userId = getUserIdFromJwt(jwt);
+
+        Check latestCheck = checkRepository.findTopByUserIdOrderByCreatedAtDesc(userId)
+                .orElseThrow(() -> new EntityNotFoundException("No checks found for this user"));
+
+        CheckDTO dto = modelMapper.map(latestCheck, CheckDTO.class);
+        return ResponseEntity.ok(dto);
+    }
+
+    private UUID getUserIdFromJwt(Jwt jwt) {
```

Comment:
Could probably move this to UserService. Would be useful for other services, too